cli/options.go

```diff
@@ -80,6 +80,10 @@ type ProjectOptions struct {
 	EnvFiles []string
 
 	loadOptions []func(*loader.Options)
+
+	// Callbacks to retrieve metadata information during parse defined before
+	// creating the project
+	Listeners map[string][]loader.Listener
```

Comment:
This makes it a bit complicated to write a listener for more than one event type. Could declare `Listeners []Listener` as `Listener` is a `func(event string, details any`, it already receives the event type, and writing a listener then is just a question of a `switch` and can select any interesting event and ignore all others.

Reply:
@ndeloof by writing the way you are suggesting, you are passing the map functionality inside every registered listener by writing in all further listeners with a switch. All registered listeners need to define the switch. When in reality with the current solution we take advantage of the map data structure to handle this. Also, when processing 1 listener, with your recomendation, you will need to iterate over all listeners, when in reality you just care for a subset of them. I can see one advantage in having a `[]Listener` instead of `map[string][]Listener` which is, in case you want to have a particular Listener that is triggered for all events